File: src/main/java/seremis/geninfusion/api/soul/IStandardSoulRegistry.java
package seremis.geninfusion.api.soul;

import net.minecraft.entity.EntityLiving;

public interface IStandardSoulRegistry {

    public void register(IStandardSoul standard);

    public IStandardSoul getStandardSoulForEntity(EntityLiving entity);

    public ISoul getSoulForEntity(EntityLiving entity);
}

File: src/main/scala/seremis/geninfusion/api/soul/IStandardSoulRegistry.scala
package seremis.geninfusion.api.soul

import net.minecraft.entity.EntityLiving

trait IStandardSoulRegistry {

    def register(standard: IStandardSoul)

    def getStandardSoulForEntity(entity: EntityLiving): Option[IStandardSoul]

    def getSoulForEntity(entity: EntityLiving): Option[ISoul]

    def getStandardSouls: Array[IStandardSoul]
}
